# Summary 
| Date   | Notes
| :----- | :-------------------------------
| 7/24   | Met w Group to discuss plans, met w Chau and Ella to try to switch GPU, familiarized naming conventions for artifacts, looked at other commands for server GPU.  
| 7/25   | Met with Chau and Ella for img+cmd, met with Group, switched GPUs, assigned GPU's, continued on img+cmd
| 7/26   | Met with Chau & Ella, implemented arg parser, uploaded models to wandb as artifacts, met with Anjali & Francsico as large group to discuss progress. 
| 7/27   | Met with Chau & Ella to go over pull request suggestions, met with Group to do updates, tried to trouble shoot dls no atribute error. 
| 7/28   | Met w Chau & Ella, debugged naming error, met with Francisco to try to fix export error. 


# Activities
* Switching to one of Server's GPU's

    * Currently not using any of them and instead running on local machine which is probably causing the error. 

   <img width="1200" alt="Screenshot 2023-07-24 at 12 24 57 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/164a73e2-04cb-4933-b921-44f139690c4d">

    * using this command in file:  torch.cuda.set_device(2) we were able to see the machine using the properly selected GPU

    <img width="1200" alt="Screenshot 2023-07-25 at 10 24 58 AM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/071f79c6-6a52-4d4b-94d4-234f4a715d6e">


    * This additionally solved our memory error and we were able to train and get results with high accuracy. 

    <img width="1200" alt="Screenshot 2023-07-25 at 10 26 19 AM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/636ff402-e3bb-439c-b3d0-503549bb217c">

* Naming Conventions for Artifacts: 

    <img width="1200" alt="Screenshot 2023-07-24 at 2 31 24 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/0c0103d8-cec3-42ea-8507-4c3378c9b866">

    * When setting up the artifact, specfy in the project = env_visibility_path. env = random/ nonrandom, visibility = (true of false for robot view, in data collection should always be false), path = wandering or perfect. 

    <img width="1200" alt="Screenshot 2023-07-24 at 2 38 56 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/22b9ee49-28bf-453d-b77c-f61c20c721b8">

    * Set the type to be "dataset" and the name to be: the date _ the number of trials _ the number of images
  

* Working with image + command: 

    * Link to 1st downoaded notebook (not current): https://github.com/daisy-abbott/ARCSLab-reports/blob/master/Assets/new-imgcmd.ipynb

    * image of terminal

    <img width="1200" alt="Screenshot 2023-07-25 at 10 35 03 AM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/d574f014-1141-4a08-b65d-4742bfebe512">

    * wandb usage charts from this run

    <img width="1200" alt="Screenshot 2023-07-25 at 10 40 48 AM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/4d8e4288-74a2-4518-9baf-424965491220">

    * New version of img+ command: https://github.com/daisy-abbott/ARCSLab-reports/blob/master/Assets/new-imgcmd1.ipynb

    * Chau's version of image + command: https://github.com/chauvuha/ARCS_Lab_Reports/blob/master/reports/2023-07-31/CmdImageClassification-test-largedataset.ipynb

    * Added arg parser to pass GPU in as an argument: 

    <img width="1200" alt="Screenshot 2023-07-26 at 1 42 59 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/bede9a41-e828-46bf-8896-83c1208b9866">

    * Added artifact run so that arcslaboratory wandb has two models that Francisco & Anjali can use in inference.py 

    <img width="1200" alt="Screenshot 2023-07-26 at 1 43 31 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/33d55217-eb60-4f4c-8148-531e40120962">

    * In wandb:
    
    <img width="1200" alt="Screenshot 2023-07-26 at 2 33 39 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/d585afcf-36df-4230-8587-971241db6556">

    * Link to most recent code: https://github.com/daisy-abbott/ARCSLab-reports/blob/master/Assets/img-cmd-v1.py

  * Changed image + command code to work with new file naming conventions 
  img of newly changed code 

* Changed function to be compatable with update naming conventions
<img width="1200" alt="Screenshot 2023-07-28 at 2 21 30 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/bcf33e6c-f212-451d-b3f0-52beee2073a0">

* Link to most recent image command code: https://github.com/daisy-abbott/ARCSLab-reports/blob/714e58bbb242d0eb6937a36ad892f3f57ec7b1ba/Assets/cur-imgcmd001.py

# Issues: 
* pkl vs. pth. I think we are going to use .pkl. (model saved as .pkl in Chau's notebook)

* change "data" to "model" in img+command at the bottome in creating artifact for folder in wandb that contains the models

* error in inference.py in downloading the model
(unreal) PS C:\Users\alnj2022\Documents\VisualStudioCode\OldenborgTraining> python inference.py C:\Users\alnj2022\Downloads\classification-resnet34-pretrained-1.pth UE_Screenshots 
Traceback (most recent call last):
  File "C:\Users\alnj2022\Documents\VisualStudioCode\OldenborgTraining\inference.py", line 95, in <module>
    main()
  File "C:\Users\alnj2022\Documents\VisualStudioCode\OldenborgTraining\inference.py", line 55, in main
    learn = load_learner(args.model)
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\alnj2022\.conda\envs\unreal\Lib\site-packages\fastai\learner.py", line 451, in load_learner
    res.dls.cpu()
    ^^^^^^^

AttributeError: 'collections.OrderedDict' object has no attribute 'dls'

* Error in running the wandering dataset and training a model on it, different naming conventions for the datasets? The wandering dataset has multiple underscores. 
  * perfect: (this one works with the img + cmd )
  <img width="1200" alt="Screenshot 2023-07-27 at 2 54 55 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/f3ac4101-0ecb-422f-9b77-8033885ab73f">

  * Wandering: this one breaks 

  <img width="1200" alt="Screenshot 2023-07-27 at 2 54 55 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/fa6c1996-9d80-4d9c-8005-d89f4201be67">

  * image of the error: 

  <img width="1200" alt="Screenshot 2023-07-27 at 2 58 40 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/99bc6fc0-e4f0-470c-abd7-085823fca635">

  * Solution: 

  <img width="1218" alt="Screenshot 2023-07-28 at 2 21 30 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/bcf33e6c-f212-451d-b3f0-52beee2073a0">


  * Unable to Export model, getting a variety of errors
    * when we took out torch.save and instead did learn.export: 
    <img width="1153" alt="Screenshot 2023-07-28 at 2 24 33 PM" src="https://github.com/daisy-abbott/ARCSLab-reports/assets/112681549/2d8040d4-d814-4d1b-8aa8-fd50038498d9">
  And when Francisco tried to download the model, he was getting an error passing it in. 

# Plans 
* Update image command code to include the greater window for forward. Convert 5 to radians and replace the zero. 
* Start surp poster
